File: Projects/放羊小分队/android/app/src/main/java/com/example/kosakai/happygoat/MyApplication.java
package com.example.kosakai.happygoat;

import android.app.Application;
import android.content.Context;
import android.widget.Toast;

import io.rong.imkit.RongIM;
import io.rong.imlib.RongIMClient;
import io.rong.imlib.model.Message;

public class MyApplication extends Application {

    private Context context = null;

    @Override
    public void onCreate() {
        super.onCreate();
        context = getApplicationContext();
        String appKey = "pwe86ga5p9e96";
        RongIM.init(context, appKey);

        RongIM.setOnReceiveMessageListener(new RongIMClient.OnReceiveMessageWrapperListener() {
            /**
             * 接收实时或者离线消息。
             * 注意:
             * 1. 针对接收离线消息时，服务端会将 200 条消息打成一个包发到客户端，客户端对这包数据进行解析。
             * 2. hasPackage 标识是否还有剩余的消息包，left 标识这包消息解析完逐条抛送给 App 层后，剩余多少条。
             * 如何判断离线消息收完：
             * 1. hasPackage 和 left 都为 0；
             * 2. hasPackage 为 0 标识当前正在接收最后一包（200条）消息，left 为 0 标识最后一包的最后一条消息也已接收完毕。
             *
             * @param message    接收到的消息对象
             * @param left       每个数据包数据逐条上抛后，还剩余的条数
             * @param hasPackage 是否在服务端还存在未下发的消息包
             * @param offline    消息是否离线消息
             * @return 是否处理消息。 如果 App 处理了此消息，返回 true; 否则返回 false 由 SDK 处理。
             */
            @Override
            public boolean onReceived(final Message message, final int left, boolean hasPackage, boolean offline) {
                return false;
            }
        });
    }

}

File: Projects/听者/Listener/app/src/main/java/com/fsyy/listener/ui/MyApplication.kt
/*
 * 项目名：Listener
 * 作者：@happy_fsyy
 * 联系我：https://github.com/happyfsyy
 * Copyright (c) 2020. Lorem ipsum dolor sit amet, consectetur adipiscing elit.
 * Morbi non lorem porttitor neque feugiat blandit. Ut vitae ipsum eget quam lacinia accumsan.
 * Etiam sed turpis ac ipsum condimentum fringilla. Maecenas magna.
 * Proin dapibus sapien vel ante. Aliquam erat volutpat. Pellentesque sagittis ligula eget metus.
 * Vestibulum commodo. Ut rhoncus gravida arcu.
 */

package com.fsyy.listener.ui

import android.app.Application
import android.content.Context
import cn.leancloud.AVOSCloud
import com.fsyy.listener.rongcloud.network.HttpUtil
import io.rong.imkit.RongIM

class MyApplication :Application(){
    companion object{
        lateinit var context:Context
        const val appkey1="appkey1"
        const val appSecret1="appSecret1"
        const val appId="appId"
        const val appKey="appKey"
        const val url="URL"
    }
    override fun onCreate() {
        super.onCreate()
        context =applicationContext
        AVOSCloud.initialize(this, appId, appKey,url)
        RongIM.init(this, appkey1)
    }
}